Source/NSLayoutAnchor.m

```diff
@@ -0,0 +1,292 @@
+                                      relatedBy: NSLayoutRelationLessThanOrEqual
+                                         toItem: [anchor item]
+                                      attribute: NSLayoutAttributeLeft
+                                     multiplier: 1.0
+                                       constant: 0.0];
+}
+
+- (NSLayoutConstraint *) constraintEqualToAnchor: (NSLayoutAnchor *)anchor constant: (CGFloat)c
+{
+  return [NSLayoutConstraint constraintWithItem: _item
+                                      attribute: NSLayoutAttributeLeft
+                                      relatedBy: NSLayoutRelationEqual
+                                         toItem: [anchor item]
+                                      attribute: NSLayoutAttributeLeft
+                                     multiplier: 1.0
+                                       constant: 0.0];
+}
+
+- (NSLayoutConstraint *) constraintGreaterThanOrEqualToAnchor: (NSLayoutAnchor *)anchor constant: (CGFloat)c
+{
+  return [NSLayoutConstraint constraintWithItem: _item
+                                      attribute: NSLayoutAttributeLeft
+                                      relatedBy: NSLayoutRelationGreaterThanOrEqual
+                                         toItem: [anchor item]
+                                      attribute: NSLayoutAttributeLeft
+                                     multiplier: 1.0
+                                       constant: c];
+}
+
+- (NSLayoutConstraint *) constraintLessThanOrEqualToAnchor: (NSLayoutAnchor *)anchor constant: (CGFloat)c;
+{
+  return [NSLayoutConstraint constraintWithItem: _item
+                                      attribute: NSLayoutAttributeLeft
+                                      relatedBy: NSLayoutRelationLessThanOrEqual
+                                         toItem: [anchor item]
+                                      attribute: NSLayoutAttributeLeft
+                                     multiplier: 1.0
+                                       constant: c];
+}
+
+- (instancetype) init
+{
+  self = [super init];
+  if (self != nil)
+    {
+      _name = nil;
+      _item = nil;
+      _hasAmbiguousLayout = NO;
+      _constraintsAffectingLayout = [[NSMutableArray alloc] init];
+    }
+  return self;
+}
+
+- (void) dealloc
+{
+  RELEASE(_name);
+  RELEASE(_item);
+  RELEASE(_constraintsAffectingLayout);
+  [super dealloc];
+}
+
+- (NSString *) name
+{
+  return _name;
+}
+
+- (id) item
+{
+  return _item;
+}
+
+- (BOOL) hasAmbiguousLayout
+{
+  return _hasAmbiguousLayout;
+}
+
+- (NSArray *) constraintsAffectingLayout
+{
+  return _constraintsAffectingLayout;
+}
+
+- (id) initWithCoder: (NSCoder *)c
+{
+  self = [super init];
+  return self;
+}
+
+- (void) encodeWithCoder: (NSCoder *)c
+{
+}
+
+- (id) copyWithZone: (NSZone *)z
+{
+  return nil;
+}
+
+@end
+
+@implementation NSLayoutDimension
+
+- (NSLayoutConstraint *) constraintEqualToConstant: (CGFloat)c
+{
+  return [NSLayoutConstraint constraintWithItem: _item
```

Comment:
According to the documentation all these methods should work on the size attribute. Either the documentation is wrong here or the code.

Reply:
The code, most likely.  Sorry, I'll correct it.

Reply:
The docs actually say "An NSLayoutConstraint object that defines a constant size for the attribute associated with this dimension anchor."

Reply:
Not sure which documentation you are using but Apple's says this:
"Returns a constraint that defines a constant size for the anchor’s size attribute."